src/lib/db/feature-toggle-store.ts

```diff
@@ -171,34 +173,40 @@ export default class FeatureToggleStore implements IFeatureToggleStore {
 
     async setLastSeen(data: LastSeenInput[]): Promise<void> {
         const now = new Date();
-        const environmentArrays = this.mapMetricDataToEnvBuckets(data);
-        try {
-            for (const env of Object.keys(environmentArrays)) {
-                const toggleNames = environmentArrays[env].sort();
-                await this.db(FEATURE_ENVIRONMENTS_TABLE)
-                    .update({ last_seen_at: now })
-                    .where('environment', env)
-                    .whereIn(
-                        'feature_name',
-                        this.db(FEATURE_ENVIRONMENTS_TABLE)
-                            .select('feature_name')
-                            .whereIn('feature_name', toggleNames)
-                            .forUpdate()
-                            .skipLocked(),
-                    );
+        const enhancedData = data.map((value) => ({
+            feature_name: value.featureName,
+            last_seen_at: now,
+            environment: value.environment,
+        }));
 
-                // Updating the toggle's last_seen_at also for backwards compatibility
-                await this.db(TABLE)
-                    .update({ last_seen_at: now })
+        try {
+            await this.db.transaction(async (trx) => {
+                // Create a CTE with the data to be inserted
+                const cte = trx('features')
```

Comment:
Is this adding a transaction on the features table?

Reply:
When I asked ChatGPT 
The features table is read very often - how can I mitigate deadlocks for transactions with more than 1000 rows
`If the features table is only used in the CTE subquery and is not being updated, the likelihood of deadlocks occurring directly due to this table is significantly reduced. In such cases, deadlocks are more likely to be related to the FEATURE_ENVIRONMENTS_METRICS_TABLE, where you are performing updates.`